core/optaplanner-constraint-streams/src/main/java/org/optaplanner/constraint/streams/bavet/common/index/Indexer.java

```diff
@@ -64,9 +67,31 @@
     Value_ remove(Object[] indexProperties, Tuple_ tuple);
 
     /**
+     * @param indexProperties never null
+     * @param visitor never null
+     */
+    void visit(Object[] indexProperties, BiConsumer<Tuple_, Value_> visitor);
+
+    /**
+     * @param indexProperties never null
+     * @return number of values across all tuples
+     */
+    default int countValues(Object[] indexProperties) {
+        AtomicInteger counter = new AtomicInteger();
+        visit(indexProperties, (tuple, value) -> counter.incrementAndGet());
+        return counter.get();
+    }
```

Comment:
Even though it is still possible to get the count of values with this method (previously `get(...).size()`), I suggest avoiding it. Sizing the maps is not nearly as beneficial as not going over the entire collection.

Reply:
Instead of doing a default method, I believe the indexer impl's should just implement it, they can do it more efficiently than this.